src/main/java/org/candlepin/audit/EventFactory.java

```diff
@@ -206,19 +197,13 @@ public Event complianceCreated(Consumer consumer, ComplianceStatus compliance) {
             ));
         }
         eventData.put("reasons", reasons);
-        try {
-            String eventDataJson = mapper.writeValueAsString(eventData);
-            // Instead of an internal db id, compliance.created events now use
-            // UUID for the 'consumerId' and 'entityId' fields, since Katello
-            // is concerned only with the consumer UUID field.
-            return new Event(Event.Type.CREATED, Event.Target.COMPLIANCE,
-                consumer.getName(), principalProvider.get(), consumer.getOwnerId(), consumer.getUuid(),
-                consumer.getUuid(), eventDataJson, null, null);
-        }
-        catch (JsonProcessingException e) {
-            log.error("Error while building JSON for compliance.created event.", e);
-            throw new IseException("Error while building JSON for compliance.created event.", e);
-        }
+
+        // Instead of an internal db id, compliance.created events now use
+        // UUID for the 'consumerId' and 'entityId' fields, since Katello
+        // is concerned only with the consumer UUID field.
+        return new Event(Event.Type.CREATED, Event.Target.COMPLIANCE,
+            consumer.getName(), principalProvider.get(), consumer.getOwnerId(), consumer.getUuid(),
```

Comment:
we should change `consumer.getOwnerId` to `consumer.getOwnerKey` here

Reply:
Good catch!! Updated